remoteworkitem/remoteworkitem_test.go

```diff
@@ -72,9 +93,95 @@ func TestGitHubIssueMapping(t *testing.T) {
 	}
 
 	for _, localWorkItemKey := range workItemMap {
-		t.Log("Mapped ", localWorkItemKey)
-		if workItem.Fields[localWorkItemKey] == nil {
+		t.Log("Mapping ", localWorkItemKey)
+		_, ok := workItem.Fields[localWorkItemKey]
+		if ok == false {
+			t.Error(fmt.Sprintf("%s not mapped", localWorkItemKey))
+		} else {
+			t.Log("Mapped to ", workItem.Fields[localWorkItemKey])
+		}
+	}
+}
+
+func TestJiraIssueMapping(t *testing.T) {
+
+	content, err := test.LoadTestData("jira_issue_mapping.json", provideRemoteJiraData)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	workItemMap := WorkItemKeyMaps[ProviderJira]
+	remoteWorkItem := RemoteWorkItem{ID: "xyz", Content: []byte(content)}
+
+	remoteWorkItemImpl := RemoteWorkItemImplRegistry[ProviderJira]
+	ji, err := remoteWorkItemImpl(remoteWorkItem)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	workItem, err := Map(ji, workItemMap)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, localWorkItemKey := range workItemMap {
+		t.Log("Mapping ", localWorkItemKey)
+		_, ok := workItem.Fields[localWorkItemKey]
+		if ok == false {
 			t.Error(fmt.Sprintf("%s not mapped", localWorkItemKey))
+		} else {
+			t.Log("Mapped to ", workItem.Fields[localWorkItemKey])
+		}
+	}
+}
+
+func TestFlattenGithubResponseMap(t *testing.T) {
+	testString, err := test.LoadTestData("github_issue_mapping.json", provideRemoteGithubData)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var nestedMap map[string]interface{}
+	err = json.Unmarshal(testString, &nestedMap)
+
+	if err != nil {
+		t.Error("Incorrect dataset ", testString)
+	}
+
+	OneLevelMap := Flatten(nestedMap)
+
+	githubKeyMap := WorkItemKeyMaps[ProviderGithub]
+
+	// Verifying if the new map is usable.
+	for k := range githubKeyMap {
+		_, ok := OneLevelMap[string(k)]
+		if ok == false && string(k) != GithubAssignee {
+			// GithubAssignee is an optional field so that being absent is fine
+			t.Error("Could not access the following key from the flattened map ", k)
+		}
+	}
+}
+
+func TestFlattenJiraResponseMap(t *testing.T) {
+	testString, err := test.LoadTestData("jira_issue_mapping.json", provideRemoteJiraData)
```

Comment:
You need to use resource.require here to specify when to run this test. See other tests for example.